### PR TITLE
all: Upgrade deps for azuresb

### LINKED ISSUE #2798: all: ambiguous import caused by Azure/go-autorest
### Describe the bug

Using `gocloud.dev` and `azure-service-bus-go` at the same time resulted ambiguous import.

In order to fix this problem, it is necessary to upgrade `azure-amqp-common-go` to `v3` and clean up the dependencies.
I am trying to fix it. #2799 

### To Reproduce

```bash
$ mkdir example && cd example/
$ go mod init example
```

main.go
```go
package main

import (
	_ "github.com/Azure/azure-service-bus-go"
	_ "gocloud.dev"
)
```

go.mod
```
module example

go 1.14

require (
	github.com/Azure/azure-service-bus-go v0.10.1
	gocloud.dev v0.19.0
)
```

```bash
$ go mod tidy
example imports
        github.com/Azure/azure-service-bus-go imports
        github.com/Azure/go-autorest/autorest/azure: ambiguous import: found package github.com/Azure/go-autorest/autorest/azure in multiple modules:
        github.com/Azure/go-autorest v12.0.0+incompatible (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/autorest/azure)
        github.com/Azure/go-autorest/autorest v0.9.3 (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest/autorest@v0.9.3/azure)
example imports
        github.com/Azure/azure-service-bus-go imports
        github.com/Azure/go-autorest/autorest/date: ambiguous import: found package github.com/Azure/go-autorest/autorest/date in multiple modules:
        github.com/Azure/go-autorest v12.0.0+incompatible (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/autorest/date)
        github.com/Azure/go-autorest/autorest/date v0.2.0 (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest/autorest/date@v0.2.0)
example imports
        github.com/Azure/azure-service-bus-go imports
        github.com/Azure/go-autorest/autorest/to: ambiguous import: found package github.com/Azure/go-autorest/autorest/to in multiple modules:
        github.com/Azure/go-autorest v12.0.0+incompatible (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/autorest/to)
        github.com/Azure/go-autorest/autorest/to v0.3.0 (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest/autorest/to@v0.3.0)
example imports
        github.com/Azure/azure-service-bus-go imports
        github.com/Azure/azure-amqp-common-go/v3/aad imports
        github.com/Azure/go-autorest/autorest/adal: ambiguous import: found package github.com/Azure/go-autorest/autorest/adal in multiple modules:
        github.com/Azure/go-autorest v12.0.0+incompatible (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/autorest/adal)
        github.com/Azure/go-autorest/autorest/adal v0.8.1 (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest/autorest/adal@v0.8.1)
example imports
        github.com/Azure/azure-service-bus-go tested by
        github.com/Azure/azure-service-bus-go.test imports
        github.com/Azure/azure-sdk-for-go/services/servicebus/mgmt/2015-08-01/servicebus imports
        github.com/Azure/go-autorest/autorest: ambiguous import: found package github.com/Azure/go-autorest/autorest in multiple modules:
        github.com/Azure/go-autorest v12.0.0+incompatible (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/autorest)
        github.com/Azure/go-autorest/autorest v0.9.3 (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest/autorest@v0.9.3)
example imports
        github.com/Azure/azure-service-bus-go tested by
        github.com/Azure/azure-service-bus-go.test imports
        github.com/Azure/azure-sdk-for-go/services/servicebus/mgmt/2015-08-01/servicebus imports
        github.com/Azure/go-autorest/autorest/validation: ambiguous import: found package github.com/Azure/go-autorest/autorest/validation in multiple modules:
        github.com/Azure/go-autorest v12.0.0+incompatible (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/autorest/validation)
        github.com/Azure/go-autorest/autorest/validation v0.2.0 (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest/autorest/validation@v0.2.0)
example imports
        github.com/Azure/azure-service-bus-go tested by
        github.com/Azure/azure-service-bus-go.test imports
        github.com/Azure/azure-sdk-for-go/services/servicebus/mgmt/2015-08-01/servicebus imports
        github.com/Azure/go-autorest/tracing: ambiguous import: found package github.com/Azure/go-autorest/tracing in multiple modules:
        github.com/Azure/go-autorest v12.0.0+incompatible (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/tracing)
        github.com/Azure/go-autorest/tracing v0.5.0 (/home/heijo/go/pkg/mod/github.com/!azure/go-autorest/tracing@v0.5.0)

```

## Expected behavior

`go mod tidy` expects to be succeed.

### Version

`gocloud.dev v0.19.0`

`go version go1.14.3 linux/amd64`

### Additional context

Using `azure-service-bus-go` downgraded to `v0.9.1` will not cause any issues.

But we had to upgrade `azure-service-bus-go` to `v0.10.1` to prevent panic. 
https://github.com/Azure/azure-service-bus-go/pull/158

### PR DESCRIPTION
**Pull request by johejo**
Fixes #2798 

Upgrade deps for azuresb
I ran the script below `internal/testing` and followed some suggestions.

### CODE REVIEW

**Comment by johejo** [comment]
@googlebot I signed it!

File: go.mod [hunk 1]
```diff
@@ -21,15 +21,16 @@ require (
 	contrib.go.opencensus.io/exporter/aws v0.0.0-20181029163544-2befc13012d0
 	contrib.go.opencensus.io/exporter/stackdriver v0.12.1
 	contrib.go.opencensus.io/integrations/ocsql v0.1.4
-	github.com/Azure/azure-amqp-common-go/v2 v2.1.0
+	github.com/Azure/azure-amqp-common-go/v3 v3.0.0
 	github.com/Azure/azure-pipeline-go v0.2.1
-	github.com/Azure/azure-sdk-for-go v30.1.0+incompatible
-	github.com/Azure/azure-service-bus-go v0.9.1
+	github.com/Azure/azure-sdk-for-go v37.1.0+incompatible
```
Comment (johejo) [inline go.mod:26]: Should I upgrade azure-sdk-for-go to the latest version? https://github.com/Azure/azure-sdk-for-go/releases/tag/v42.3.0

Comment (vangent) [reply go.mod:26]: Sure, might as well.

Comment (vangent) [reply go.mod:26]: Let's do it in a separate PR though.

**Pull request merged**